packages/client/hmi-client/src/composables/project.ts

```diff
@@ -0,0 +1,224 @@
+import { IProject } from '@/types/Project';
+import { Component, Ref, readonly, shallowRef } from 'vue';
+import * as ProjectService from '@/services/project';
+import * as CodeService from '@/services/code';
+import * as ArtifactService from '@/services/artifact';
+import * as DatasetService from '@/services/dataset';
+import * as ModelService from '@/services/model';
+import { AssetType } from '@/types/Types';
+import DatasetIcon from '@/assets/svg/icons/dataset.svg?component';
+
+const TIMEOUT_MS = 1000;
+
+const activeProject = shallowRef<IProject | null>(null);
+const allProjects = shallowRef<IProject[] | null>(null);
```

Comment:
What are the implications of using shallowRefs? Does this mean that any direct changes to the data structure (e.g. changing the description) are not reflected until we refetch the data again? 

It would be good if you can add some comments on the expected usage patterns to make the expectations more explicit.

Reply:
Yes, that is correct, changes to the data description would not be reflected until the data is refreshed. Note that this is the existing paradigm as well, and in the previous implementation, `project` and `projects` in `App.vue` were also shallowRefs. 

However the idea here is that all the functions in `useProjects` guarantee that any changes to `activeProject` will be propagated to all of its users, so any changes to `activeProject` or its assets should be made through `useProjects` functions if you want to guarantee that those changes are propagated.

Reply:
Okay, can you just add a brief comment to the code as such? Just so the readers know explicitly what are the intentions and what to expect.

Reply:
Done

Reply:
You can mark as resolved yourself if it is done.